tasklist/client/src/v2/TaskDetailsLayout/AssignButton/index.tsx

```diff
@@ -143,6 +155,12 @@ const AssignButton: React.FC<Props> = ({id, assignee, currentUser}) => {
     }
   };
 
+  useEffect(() => {
+    if (taskState === 'ASSIGNING') {
+      setAssignmentStatus('assigning');
+    }
+  }, [taskState, setAssignmentStatus]);
```

Comment:
Can't we simply do this on the initial value of `useState`? It's generally not a good idea to use `setState` inside a `useEffect`

With the if is less bad but it can easily create the conditions for bugs if someone changes this in the future

Some docs on it:
- https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
- https://kentcdodds.com/blog/dont-sync-state-derive-it